File: 02.Filas_Prioridade/AppSorts.java
import java.util.Random;

public class AppSorts {

    public static void main(String args[]) {
        Random r = new Random();
        final int MAX = 100000;

        for (int max = 500; max <= MAX; max += 500) {
            Integer[] data = new Integer[max];
            for (int i = 0; i < data.length; i++)
                data[i] = r.nextInt(data.length * 10);

            //Mergesort<Integer> s = new Mergesort<>();
            Quicksort<Integer> s = new Quicksort<>();
            // MaxHeap<Integer> s = new MaxHeap<data.length>();

            long start = System.nanoTime();
            s.sort(data);
            long end = System.nanoTime();

            double tempo = (end - start) / 1e9; // 1.000.000 (para transformar em ms)
            System.out.printf("%d %.7f\n", max, tempo);
        }
    }
}

File: 02.Filas_Prioridade/app_sorts.py
from mergesort import Mergesort
from quicksort import Quicksort
from maxheap import MaxHeap
from time import process_time

import random

if __name__ == "__main__":

    MAX = 50000

    for n in range(500,MAX,500):
        data = [random.randint(1,n*10) for x in range(n)]
        #print(data)

        s = Quicksort()
        #s = Mergesort()
        #s = MaxHeap()

        start = process_time()

        s.sort(data)

        end = process_time()

        t = end-start
        print(f"{n} {t}")

